autorest/app.ts

```diff
@@ -181,6 +187,56 @@ async function clearTempData() {
   await Promise.all(all);
 }
 
+async function configurationSpecifiedVersion(selectedVersion: any) {
+  try {
+    // we can either have a selectedVerison object or a path. See if we can find the AutoRest API
+    const autorestApi = await resolveEntrypoint(typeof selectedVersion === 'string' ? selectedVersion : await selectedVersion.modulePath, 'main');
+
+    // things we need in the sandbox.
+    const sandbox = {
+      require, console, rfs: {
+        EnumerateFileUris: async (folderUri: string): Promise<Array<string>> => {
+          return EnumerateFiles(folderUri, ['readme.md']);
+        },
+        ReadFile: async (uri: string): Promise<string> => {
+          return ReadUri(uri);
+        },
+        WriteFile: async (uri: string, content: string): Promise<void> => {
+          //return WriteString(uri, content);
+        }
+      },
+      cfgfile: ResolveUri(cwd, args.configFileOrFolder || '.'),
+      switches: args
+    };
+
+    // *sigh* ... there's a bug in most versions of autorest-core that to use the API you have to 
+    // have the current directory set to the package location. We'll fix this in the future versions.
+    process.chdir(dirname(autorestApi));
+    const configSpecifiedVersion = await vm.runInNewContext(`
```

Comment:
I load whatever version of the autorest-core (into a sandbox) we can get our hands on, just to let it do the parsing of the configuration. 

If the configuration contains a `version` value, we substitute out that for the version that we assumed.